### ISSUE TITLE
Support for 3.6+ $expr in queries

### ISSUE THREAD

**Issue by chrisdejager** [doctrine/mongodb-odm#1824]
I would love to be able to compare fields with each other in a query without needing to fall back on javascript in the where operator.

Since MongoDB 3.6+ we can use expressions within queries: https://docs.mongodb.com/v3.6/reference/operator/query/expr/
Example: db.monthlyBudget.find( { $expr: { $gt: [ "$spent" , "$budget" ] } } )

Using $expr instead of javascript would obviously perform better.

Is this planned for a future release?

Thanx!
Chris

**Comment by alcaeus** [comment]
Yes - we've got some catching up on those. Current focus is on ODM 2.0, which will (finally) get rid of the legacy driver and thus allow us to start working on new features introduced in more recent MongoDB server versions. I'm triaging this for the 2.x series as 2.0 won't bring many new features of that kind - I'd rather start following up with more rapid minor releases in short succession.

**Comment by chrisdejager** [comment]
I expected something like that :) Thanx for the heads up.

Thanks a million for this library!
